jdaviz/configs/specviz/plugins/viewers.py

```diff
@@ -545,8 +545,24 @@ def _plot_uncertainties(self):
                 self.figure.marks = list(self.figure.marks) + [error_line_mark]
 
     def set_plot_axes(self):
-        # Set axes labels for the spectrum viewer
+        # Set y axes labels for the spectrum viewer
+        y_display_unit = self.state.y_display_unit
+        y_unit = u.Unit(y_display_unit) if y_display_unit else u.dimensionless_unscaled
+
+        # default to Flux Density for flux density or uncaught types
         flux_unit_type = "Flux density"
```

Comment:
this might be a little easier to read (and catch bugs like previously happened), if this was in an else at the end of the elifs 🤷‍♂️

Reply:
but it needs to be in an else, otherwise it will always be applied

Reply:
oh whoops haha